core/src/test/java/kafka/testkit/KafkaClusterTestKit.java

```diff
@@ -373,13 +373,19 @@ public void startup() throws ExecutionException, InterruptedException {
 
     /**
      * Wait for a controller to mark all the brokers as ready (registered and unfenced).
+     * And also wait for the metadata cache up-to-date in each broker server.
      */
     public void waitForReadyBrokers() throws ExecutionException, InterruptedException {
         // We can choose any controller, not just the active controller.
         // If we choose a standby controller, we will wait slightly longer.
         ControllerServer controllerServer = controllers.values().iterator().next();
```

Comment:
@dengziming , yes, you are right, it can still work well if we remove `waitForReadyBrokers`. But I think it's still useful to keep it, because if we have test failed with "metadata cache not update in time", we can't know if all the brokers are ready (registered and unfenced) or not. That is, with the `controller.waitForReadyBrokers()`, we can make the tests better for troubleshooting in the future. WDYT?

Reply:
Yeah, this makes sense.